pymongo_inmemory/context.py

```diff
@@ -127,6 +127,7 @@ def __init__(
         )
         self.archive_folder = mkdir_ifnot_exist(self.download_folder, self.url_hash)
         self.extracted_folder = mkdir_ifnot_exist(self.extract_folder, self.url_hash)
+        self.storage_engine = "ephemeralForTest"
```

Comment:
Maybe the fallback option should be `inMemory`?

Reply:
Yes, I like `inMemory`. I'm not sure though what you mean by "fallback option". When would you apply the fallback option?
In this PR, the default setting (`ephemeralForTest`) doesn't change. It is just made configurable using the `Context` object.
If someone wants to not configure any specific storage engine or setting it to `inMemory`, that's now also possible. Of course, if you do this you need to consider the cases you mentioned (writing data to disk, permissions etc.).

Reply:
I tested it a bit with `inMemory`, but it doesn't work as it looks like this option only available in the Enterprise version (https://www.mongodb.com/docs/manual/core/inmemory/)
> Starting in MongoDB Enterprise version 3.2.6, the in-memory storage engine is part of general availability (GA) in the 64-bit builds.